### PR TITLE
UCP/API/DOC: enum ucp_ep_close_mode

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
This API/DOC change is separated from https://github.com/openucx/ucx/pull/1822

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4506/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp @bbenton pls take a look

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -217,17 +217,24 @@ enum ucp_ep_params_flags_field {
...
                                               completed with
-                                              @ref UCS_ERR_CANCELED error. */
+                                              @ref UCS_ERR_CANCELED error.
+                                              @note This mode may cause
+                                              transport level errors on remote
+                                              side, so it requires set
+                                              @ref UCP_ERR_HANDLING_MODE_PEER
+                                              for all endpoints created on
+                                              both (local and remote) sides to
+                                              avoid undefined behaviour */
```
Comment (bbenton) [inline src/ucp/api/ucp.h]: Minor cleanup:
  pls use American English spelling for 'behavior'
  pls end with a period since it's a complete sentence

**Comment by bbenton** [comment]
:+1:  for doc/API

**Pull request merged**